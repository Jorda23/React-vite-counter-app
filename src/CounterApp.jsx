import { useState } from 'react';
import PropTypes from 'prop-types'



export const CounterApp = ( { value } ) => {

    // console.log('render');
    //API...... FETCH.........

    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => {
        setCounter( counter + 1); //el setCounter cambia el valor de counter que es el valor de value
        // setCounter( (c) => c + 1);
    }

    const handleSubtract = () => {
        setCounter( counter - 1); 
    }

    const initialReset = () => {
        setCounter( value ); 
    }

    return (
    <>
      <h1>CounterApp</h1>
      <h2> { counter } </h2>

      <button onClick={ handleAdd }> +1 </button>
      <button onClick={ handleSubtract }>-1</button>
      <button onClick={  initialReset }>Reset</button>
    </>
  );
}

CounterApp.propTypes = {
    value: PropTypes.number,
}

