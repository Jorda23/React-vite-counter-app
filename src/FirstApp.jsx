import PropTypes from 'prop-types'

export const FirstAppWatch = ( { title, subTitle, name} ) => {
  if(!title){
   throw new Error ('The title not exist')
  }
  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )  
}

FirstAppWatch.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
}

FirstAppWatch.defaultProps = {
  name: "My name is Jordan",
  subTitle: "There is not subtitle",
  title: "There is not title, thanks for attention"
}