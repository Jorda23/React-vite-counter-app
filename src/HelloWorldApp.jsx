// function App() {
//     return (<h1>Hola mundo XDXDXD</h1>);
// };

// export default App;

export const HelloWorldApp = () => {
    return (
        <div>Hola mundo XDXDXD</div>
    );
}

