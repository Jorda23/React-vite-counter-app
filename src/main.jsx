import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
// import { FirstAppWatch } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';
import './Styles.css'


ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={78}/>
        {/* <HelloWorldApp /> */}
        {/* <FirstAppWatch /> */}
    </React.StrictMode>
);

