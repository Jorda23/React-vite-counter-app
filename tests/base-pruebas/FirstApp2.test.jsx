import { render, screen } from "@testing-library/react";
import { FirstAppWatch } from "../../src/FirstApp"

describe('Tets in <FirstAppWatch />', () => {
    
    const title= 'Hello, I am Goku';
    const subTitle=  'I am a subTitle';

    test('debe hacer math con el snapshot', () => {
      const { container } = render( <FirstAppWatch title={title} />);
      expect( container ).toMatchSnapshot();
    });

    test('Debe de mostar el mensaje de, "Hello, I am Goku"', () => {
        render( <FirstAppWatch title={ title} />);
        expect( screen.getByText(title)).toBeTruthy();
    });

    test('Debe de mostar el titulo en un h1', () => {
        render( <FirstAppWatch title={ title }/>);
        expect( screen.getByRole('heading', { level: 1}).innerHTML).toContain(title);
    });

    test('Debe de mostrar el subtitulo enviado por props', () => {
        render( 
            <FirstAppWatch 
                title={ title }
                subTitle={ subTitle }
            />
        );

        expect( screen.getAllByText(subTitle).length ).toBe(1);
    });
    
    
    
});
