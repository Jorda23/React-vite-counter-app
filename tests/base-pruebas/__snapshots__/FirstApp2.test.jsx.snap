// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tets in <FirstAppWatch /> debe hacer math con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello, I am Goku
  </h1>
  <p>
    There is not subtitle
  </p>
  <p>
    My name is Jordan
  </p>
</div>
`;
